sdk/appconfiguration/azure-data-appconfiguration/swagger/src/main/java/AppConfigCustomization.java

```diff
@@ -0,0 +1,146 @@
+        classCustomization.getMethod("setLabel")
+            .getJavadoc()
+            .setDescription("Set the label property: Filters {@link ConfigurationSetting} by their label field.")
+            .setReturn("the {@link KeyValueFilter} object itself.");
+        classCustomization.getMethod("getKey")
+            .getJavadoc()
+            .setDescription("Get the key property: Filters {@link ConfigurationSetting} by their key field.");
+        classCustomization.getMethod("getLabel")
+            .getJavadoc()
+            .setDescription("Get the label property: Filters {@link ConfigurationSetting} by their label field.");
+
+        classCustomization = classCustomization.customizeAst(
+            ast -> ast.getClassByName("KeyValueFilter")
+                       .get()
+                       .getDefaultConstructor().get()
+                       .remove())
+                                 .removeMethod("setKey")
+                                 .rename("SnapshotSettingFilter");
+
+        classCustomization.addConstructor(joinWithNewline(
+            "public SnapshotSettingFilter(String key) {",
+            "   this.key = key;",
+            "}"))
+            .getJavadoc()
+            .setDescription("Constructor of {@link SnapshotSettingFilter}.")
+            .setParam("key", "The key value to set.");
+    }
+
+    private void customizeKeyValueFields(ClassCustomization classCustomization) {
+        classCustomization.addImports("java.util.Locale;");
+        classCustomization.addImports("com.azure.data.appconfiguration.ConfigurationAsyncClient;");
+
+        classCustomization = classCustomization.removeMethod("values");
+
+        classCustomization.getConstant("KEY")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getKey()} from the service.");
+
+        classCustomization.getConstant("LABEL")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLabel()} from the service.");
+
+        classCustomization.getConstant("VALUE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getValue()} from the service.");
+
+        classCustomization.getConstant("CONTENT_TYPE")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getContentType()} from the service.");
+
+        classCustomization.getConstant("LAST_MODIFIED")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getLastModified()} from the service.");
+
+        classCustomization.getConstant("TAGS")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getTags()} from the service.");
+
+        classCustomization.getConstant("ETAG")
+            .getJavadoc()
+            .setDescription("Populates the {@link ConfigurationSetting#getETag()} from the service.");
```

Comment:
These should be possible with Swagger transforms modifying the `description` of the enum.

```
directive:
  - from: swagger-document
    where: $.definitions.KeyVauleFields
    transform: >
      $.items["x-ms-enum"].values = [
        {
          "value": "key",
          "name": "key",
          "description": "Populates the 'key' from the service.
        },
        ...
      ];
```

Where `value` is the value of the expandable string enum, in this case for `IS_READ_ONLY` below it would be `locked`, and `name` is the name of the constant. I would stick with using the JSON names of the configuration properties in case there is any issues with descriptions including link tags.